### ISSUE TITLE
Wrong warning "Use initializing formals when possible" in factory ctors

### ISSUE THREAD

**Issue by deakjahn** [dart-lang/linter#3065]
**Describe the issue**
I get a "Use initializing formals when possible" warning in the code below. If I follow the warning and let the code to be refactored, it will become an error: "Initializing formal parameters can't be used in factory constructors."

**To Reproduce**
```
late Locale locale;
late String path;

factory L10n.create(Locale locale, String path) {
  _instance.locale = locale;
  _instance.path = path;
  return _instance;
}
```

**Comment by pq** [comment]
Oof yeah, sorry. This was a false positive (also reported in https://github.com/dart-lang/linter/issues/2441).

Closing this one since it was fixed in https://github.com/dart-lang/linter/pull/2854 and will be available after our next push to the SDK.

Sorry for the inconvenience!
